gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -198,6 +199,17 @@ protected void runJob(String jobName, Properties jobProps, JobState jobState, Li
     // Turn off speculative execution
     this.job.setSpeculativeExecution(false);
 
+    // Delete any staging directories that already exist
```

Comment:
AbstractJobLauncher.runWorkUnits() runs in the map tasks, which I'm trying to avoid. Since multiple writers write to the same directories, and the writers could be in different map tasks, it's possible one task could delete the data of another task.

Also, I added it directly to MRJobLauncher because I JobLaunchUtils.cleanStagingData() requires a filesystem connection.